agent/lib/kontena/workers/weave_worker.rb

```diff
@@ -48,19 +48,21 @@ def on_dns_add(topic, event)
     # @param [String] topic
     # @param [Docker::Event] event
     def on_container_event(topic, event)
-      if event.status == 'start'
+      if network_adapter.router_image?(event.from)
+        if event.status == 'start' || event.status == 'restart'
```

Comment:
In case `docker restart xyz` you will actually get both `start` and `restart` events:
```
2017-04-26T12:42:16.871540235+03:00 container start 5266dd1ce9128826f16dace277110b263f79587519608c71cc21df6d0b19e450 (image=nginx:1-alpine, io.kontena.container.deploy_rev=2017-04-21 10:41:25 UTC, io.kontena.container.name=nginx-1, io.kontena.container.overlay_cidr=10.81.128.63/16, io.kontena.container.overlay_network=kontena, io.kontena.container.pod=nginx-1, io.kontena.container.service_revision=1, io.kontena.container.type=container, io.kontena.grid.name=e2e, io.kontena.service.id=58f9e1cf6cc0b100090004a9, io.kontena.service.instance_number=1, io.kontena.service.name=nginx, io.kontena.stack.name=null, name=nginx-1)
2017-04-26T12:42:16.871768064+03:00 container restart 5266dd1ce9128826f16dace277110b263f79587519608c71cc21df6d0b19e450 (image=nginx:1-alpine, io.kontena.container.deploy_rev=2017-04-21 10:41:25 UTC, io.kontena.container.name=nginx-1, io.kontena.container.overlay_cidr=10.81.128.63/16, io.kontena.container.overlay_network=kontena, io.kontena.container.pod=nginx-1, io.kontena.container.service_revision=1, io.kontena.container.type=container, io.kontena.grid.name=e2e, io.kontena.service.id=58f9e1cf6cc0b100090004a9, io.kontena.service.instance_number=1, io.kontena.service.name=nginx, io.kontena.stack.name=null, name=nginx-1)
```